### PR TITLE
update composer dependencies

### PR DESCRIPTION
**Pull request by h-bragg**
- Add PHP 7.2
- 🆙 `graze/standards` to v2
- 🆙 `phpunit` up to v7
- 🆙 `symfony` with v4

### CODE REVIEW

**Comment by h-bragg** [comment]
@biggianteye Thoughts on new baseline config for testing multiple versions locally? (also supports nightly php versions)

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,4 @@
 /vendor/
-composer.lock
 /tests/report/
+
```
Comment (biggianteye) [inline .gitignore:3]: blank line

File: Makefile [hunk 2]
```diff
@@ -16,17 +19,25 @@ PREFER_LOWEST ?=
...
-	make 'composer-install --optimize-autoloader --ignore-platform-reqs'
+	make 'composer-install --optimize-autoloader'
 
 build-update: ## Update the dependencies
-	make 'composer-update --optimize-autoloader --ignore-platform-reqs ${PREFER_LOWEST}'
+	make 'composer-update --optimize-autoloader ${PREFER_LOWEST}'
+
+ensure-composer-file: # Update the composer file
+ifeq (${OS},Darwin)
+	sed -E -e 's/"php": "[0-9\.]+"/"php": "${PHP_VER}"/' -i '' composer.json
```
Comment (biggianteye) [inline Makefile:29]: As discussed in slack, this editing is a bit icky but I understand why it was done.

File: Makefile [hunk 3]
```diff
@@ -16,17 +19,25 @@ PREFER_LOWEST ?=
...
+else
+	sed -r -e's/"php": "[0-9\.]+"/"php": "${PHP_VER}"/' -i'' composer.json
```
Comment (biggianteye) [inline Makefile:31]: I wish macOS provide GNU tools by default rather than the gross BSD ones.

File: Makefile [hunk 4]
```diff
@@ -49,17 +60,15 @@ test-lowest: ## Test using the lowest possible versions of the dependencies
...
-	${MAKE} DOCKER_RUN="${DOCKER_RUN_BASE} hhvm/hhvm:latest" build-update test
-	${MAKE} DOCKER_RUN="${DOCKER_RUN_BASE} php:5.5-alpine" PREFER_LOWEST=--prefer-lowest build-update test
-	${MAKE} DOCKER_RUN="${DOCKER_RUN_BASE} php:5.6-alpine" PREFER_LOWEST=--prefer-lowest build-update test
-	${MAKE} DOCKER_RUN="${DOCKER_RUN_BASE} php:7.0-alpine" PREFER_LOWEST=--prefer-lowest build-update test
-	${MAKE} DOCKER_RUN="${DOCKER_RUN_BASE} php:7.1-alpine" PREFER_LOWEST=--prefer-lowest build-update test
-	${MAKE} DOCKER_RUN="${DOCKER_RUN_BASE} hhvm/hhvm:latest" PREFER_LOWEST=--prefer-lowest build-update test
+	${MAKE} PHP_VER="5.6" build-update test
+	${MAKE} PHP_VER="7.0" build-update test
+	${MAKE} PHP_VER="7.1" build-update test
+	${MAKE} PHP_VER="7.2" build-update test
```
Comment (biggianteye) [inline Makefile:71]: Cleaner in both the code and the command being executed. 👍

**Review by biggianteye — APPROVED** [review]

**Pull request merged**